envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoyListenersFactory.kt

```diff
@@ -0,0 +1,304 @@
+package pl.allegro.tech.servicemesh.envoycontrol.snapshot
+
+import com.google.protobuf.BoolValue
+import com.google.protobuf.Duration
+import com.google.protobuf.Struct
+import com.google.protobuf.Value
+import io.envoyproxy.envoy.api.v2.Listener
+import io.envoyproxy.envoy.api.v2.core.Address
+import io.envoyproxy.envoy.api.v2.core.AggregatedConfigSource
+import io.envoyproxy.envoy.api.v2.core.ApiConfigSource
+import io.envoyproxy.envoy.api.v2.core.ConfigSource
+import io.envoyproxy.envoy.api.v2.core.GrpcService
+import io.envoyproxy.envoy.api.v2.core.Http1ProtocolOptions
+import io.envoyproxy.envoy.api.v2.core.SocketAddress
+import io.envoyproxy.envoy.api.v2.listener.Filter
+import io.envoyproxy.envoy.api.v2.listener.FilterChain
+import io.envoyproxy.envoy.config.accesslog.v2.FileAccessLog
+import io.envoyproxy.envoy.config.filter.accesslog.v2.AccessLog
+import io.envoyproxy.envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager
+import io.envoyproxy.envoy.config.filter.network.http_connection_manager.v2.HttpFilter
+import io.envoyproxy.envoy.config.filter.network.http_connection_manager.v2.Rds
+import pl.allegro.tech.servicemesh.envoycontrol.groups.Group
+import com.google.protobuf.Any as ProtobufAny
+import io.envoyproxy.envoy.config.filter.http.header_to_metadata.v2.Config as HeaderToMetadataConfig
+
+typealias HttpFilterFactory = (node: Group) -> HttpFilter?
+
+class EnvoyDefaultFilters(private val snapshotProperties: SnapshotProperties) {
+    val defaultServiceTagFilterRules = serviceTagFilterRules()
```

Comment:
Feels like the listeners implementation contains specific details of a feature such as service tags handling and the default settings for that and probably other things too, which should be configured in their dedicated package. Please suggest a more package-oriented approach. Now it feels this is a global object with knowledge about a lot of things, that should be pluggable.